Cannot run npm install on Mac due to sharp dependency

### What version of `starlight` are you using?

34.3

### What version of `astro` are you using?

5.6.1

### What package manager are you using?

npm

### What operating system are you using?

Mac

### What browser are you using?

Chrome

### Describe the Bug

Expected: I should be able to create a new project, run `npm install`, and then run `npm run dev` without errors.
Actual: Running `npm install` on a fresh project fails due to the `sharp` dependency.

Steps to reproduce:
1. `npm create astro@latest -- --template starlight`
2. cd into the new directory
3. `npm install`

Output:
```
npm ERR! code 1
npm ERR! path ~/code/newproject.com/node_modules/sharp
npm ERR! command failed
npm ERR! command sh -c (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
npm ERR! sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.14.5/libvips-8.14.5-darwin-arm64v8.tar.br
npm ERR! sharp: Please see https://sharp.pixelplumbing.com/install for required dependencies
npm ERR! sharp: Installation error:

npm ERR! A complete log of this run can be found in: ~/.npm/_logs/2025-06-24T23_39_50_529Z-debug-0.log
```

After a bit of research, I was able to overcome this by first running `npm install sharp@0.34.2 --save`, and then I was able to successfully run `npm install` and `npm run dev`.

### Link to Minimal Reproducible Example

See notes above.

### Participation

- [ ] I am willing to submit a pull request for this issue.

Just reproduced it again as a sanity check. Here's the exact commands and output.

```
richardmorgan@Richards-MacBook-Pro-2 GitHub % npm create astro@latest -- --template starlight

 astro   Launch sequence initiated.

   dir   Where should we create your new project?
         ./starlight-sharp-bug
      ◼  tmpl Using starlight as project template

  deps   Install dependencies?
         Yes

   git   Initialize a new git repository?
         Yes

      ▲  error Error
      ▲  error Dependencies failed to install, please run npm install to install them manually after setup.
      ✔  Project initialized!
         ■ Template copied
         ■ Dependencies installed
         ■ Git initialized

  next   Liftoff confirmed. Explore your project!

         Enter your project directory using cd ./starlight-sharp-bug 
         Run npm run dev to start the dev server. CTRL+C to stop.
         Add frameworks like react or tailwind using astro add.

         Stuck? Join us at https://astro.build/chat

╭─────╮  Houston:
│ ◠ ◡ ◠  Good luck out there, astronaut! 🚀
╰─────╯
richardmorgan@Richards-MacBook-Pro-2 GitHub % cd starlight-sharp-bug 
richardmorgan@Richards-MacBook-Pro-2 starlight-sharp-bug % npm install
npm ERR! code 1
npm ERR! path /Users/richardmorgan/code/GitHub/starlight-sharp-bug/node_modules/sharp
npm ERR! command failed
npm ERR! command sh -c (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
npm ERR! sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.14.5/libvips-8.14.5-darwin-arm64v8.tar.br
npm ERR! sharp: Please see https://sharp.pixelplumbing.com/install for required dependencies
npm ERR! sharp: Installation error:

npm ERR! A complete log of this run can be found in: /Users/richardmorgan/.npm/_logs/2025-06-25T22_20_02_078Z-debug-0.log
richardmorgan@Richards-MacBook-Pro-2 starlight-sharp-bug % npx astro info
Need to install the following packages:
astro@5.10.1
Ok to proceed? (y) y
3:20:25 PM [vite] (ssr) Error when evaluating SSR module /Users/richardmorgan/code/GitHub/starlight-sharp-bug/astro.config.mjs: Cannot find module 'astro/config' imported from '/Users/richardmorgan/code/GitHub/starlight-sharp-bug/astro.config.mjs'
      at fetchModule (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:46813:19)
      at RunnableDevEnvironment.fetchModule (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:47904:12)
      at fetchModule (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:47858:21)
      at handleInvoke (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:38973:28)
      at EventEmitter.listenerForInvokeHandler (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:39046:25)
      at EventEmitter.emit (node:events:514:28)
      at Object.send (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:25190:40)
      at Object.invoke (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/module-runner.js:621:34)
      at Object.invoke (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/module-runner.js:682:34)
      at SSRCompatModuleRunner.getModuleInformation (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/module-runner.js:1205:94)
[astro] Unable to load your Astro config

Cannot find module 'astro/config' imported from '/Users/richardmorgan/code/GitHub/starlight-sharp-bug/astro.config.mjs'
  Location:
    /Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:46813:19
  Stack trace:
    at fetchModule (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:46813:19)
    at fetchModule (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:47858:21)
    at EventEmitter.listenerForInvokeHandler (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:39046:25)
    at Object.send (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/chunks/dep-DBxKXgDP.js:25190:40)
    at Object.invoke (file:///Users/richardmorgan/.npm/_npx/aa98e6899c6baff3/node_modules/vite/dist/node/module-runner.js:682:34)
richardmorgan@Richards-MacBook-Pro-2 starlight-sharp-bug % cat /Users/richardmorgan/.npm/_logs/2025-06-25T22_20_02_078Z-debug-0.log
cat: /Users/richardmorgan/.npm/_logs/2025-06-25T22_20_02_078Z-debug-0.log: No such file or directory
```

Unfortunately, `npx astro info` fails, I assume due to `npm install` failing to install dependencies. I tried installing the pinned version of starlight manually, and that fails as well due to the `sharp` dependency:
```
npm install @astrojs/starlight@0.34.4
```

Here's the end of the npm log:
```

2818 timing build:link:node_modules/sitemap Completed in 2ms
2819 timing build:link:node_modules/typescript Completed in 2ms
2820 timing build:link:node_modules/tsconfck Completed in 2ms
2821 timing build:link:node_modules/vite Completed in 2ms
2822 timing build:link Completed in 3ms
2823 info run sharp@0.32.6 install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
2824 info run sharp@0.33.5 install node_modules/astro/node_modules/sharp node install/check
2825 info run sharp@0.33.5 install { code: 0, signal: null }
2826 timing build:run:install:node_modules/astro/node_modules/sharp Completed in 4312ms
2827 info run sharp@0.32.6 install { code: 1, signal: null }
2828 timing reify:rollback:createSparse Completed in 877ms
2829 timing reify:rollback:retireShallow Completed in 0ms
2830 timing command:install Completed in 28011ms
2831 verbose stack Error: command failed
2831 verbose stack     at ChildProcess.<anonymous> (/Users/richardmorgan/.nvm/versions/node/v20.10.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
2831 verbose stack     at ChildProcess.emit (node:events:514:28)
2831 verbose stack     at maybeClose (node:internal/child_process:1105:16)
2831 verbose stack     at Socket.<anonymous> (node:internal/child_process:457:11)
2831 verbose stack     at Socket.emit (node:events:514:28)
2831 verbose stack     at Pipe.<anonymous> (node:net:337:12)
2832 verbose pkgid sharp@0.32.6
2833 verbose cwd /Users/richardmorgan/code/GitHub/starlight-sharp-bug
2834 verbose Darwin 23.6.0
2835 verbose node v20.10.0
2836 verbose npm  v10.2.3
2837 error code 1
2838 error path /Users/richardmorgan/code/GitHub/starlight-sharp-bug/node_modules/sharp
2839 error command failed
2840 error command sh -c (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
2841 error sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.14.5/libvips-8.14.5-darwin-arm64v8.tar.br
2841 error sharp: Please see https://sharp.pixelplumbing.com/install for required dependencies
2842 error sharp: Installation error:
2843 verbose exit 1
2844 timing npm Completed in 28140ms
2845 verbose unfinished npm timer reify 1750891084414
2846 verbose unfinished npm timer reify:build 1750891106277
2847 verbose unfinished npm timer build 1750891106278
2848 verbose unfinished npm timer build:deps 1750891106278
2849 verbose unfinished npm timer build:run:install 1750891106286
2850 verbose unfinished npm timer build:run:install:node_modules/sharp 1750891106286
2851 verbose code 1
2852 error A complete log of this run can be found in: /Users/richardmorgan/.npm/_logs/2025-06-25T22_38_04_284Z-debug-0.log
```

--- a/docs/package.json
+++ b/docs/package.json
@@ -22,7 +22,7 @@
     "@types/culori": "^2.1.1",
     "astro": "^5.6.1",
     "culori": "^4.0.1",
-    "sharp": "^0.32.5"
+    "sharp": "^0.34.2"
   },
   "devDependencies": {
     "@playwright/test": "^1.45.0",

--- a/examples/basics/package.json
+++ b/examples/basics/package.json
@@ -13,6 +13,6 @@
   "dependencies": {
     "@astrojs/starlight": "^0.34.4",
     "astro": "^5.6.1",
-    "sharp": "^0.32.5"
+    "sharp": "^0.34.2"
   }
 }

--- a/examples/markdoc/package.json
+++ b/examples/markdoc/package.json
@@ -15,6 +15,6 @@
     "@astrojs/starlight": "^0.34.4",
     "@astrojs/starlight-markdoc": "^0.4.0",
     "astro": "^5.6.1",
-    "sharp": "^0.32.5"
+    "sharp": "^0.34.2"
   }
 }

--- a/examples/tailwind/package.json
+++ b/examples/tailwind/package.json
@@ -15,7 +15,7 @@
     "@astrojs/starlight-tailwind": "^4.0.1",
     "@tailwindcss/vite": "^4.0.7",
     "astro": "^5.6.1",
-    "sharp": "^0.32.5",
+    "sharp": "^0.34.2",
     "tailwindcss": "^4.0.7"
   }
 }

